lib/connection.js

```diff
@@ -328,15 +362,22 @@ class Connection extends EventEmitter {
     this.emit('_timeout');
   }
 
+  // Prepare a JSTP message to be sent over this connection
+  //   message - a message to prepare
+  //
+  _prepareMessage(message) {
+    return serde.stringify(message);
+  }
+
   // Send a JSTP message over this connection
-  //   message - a message to send
+  //   preparedMessage - a prepared message to send
+  //   message - a message object (used for development events thus optional)
   //
-  _send(message) {
-    const data = serde.stringify(message);
-    this.transport.send(data);
+  _send(preparedMessage, message) {
```

Comment:
Why can't we make 2 send methods `_send(message)` and `_sendPrepared(preparedMessage)` and always parse prepared message for debug mode events in the latter method with the former method only calling the latter one after preparing a message? This will avoid a lot of boilerplate code and only affect debug mode performance which is negligible.